{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/tests.py

```diff
@@ -90,8 +93,37 @@ def test_password_reset(test_user, client):
 def test_user_token_gets_created_from_signal(test_user):
     assert test_user.auth_token
 
+{%- if cookiecutter.include_mobile == 'y' %}
+@pytest.mark.django_db
+def test_user_password_reset_request(test_user):
+    client = Client()
+    res = client.get(f"/api/password/reset/code/{test_user.email}/", **JSON_RQST_HEADERS)
+    assert res.status_code == 204
+    assert UserResetPasswordCode.objects.count()
+
 
 @pytest.mark.django_db
+def test_user_can_change_password_with_code(test_user):
+    client = Client()
+    res = client.post("/api/login/", {"email": test_user.email, "password": "testing123"}, **JSON_RQST_HEADERS)
+    assert res.status_code == 200
+    assert res.json()
+    res = client.get(f"/api/password/reset/code/{test_user.email}/", **JSON_RQST_HEADERS)
+    assert res.status_code == 204
+    assert UserResetPasswordCode.objects.count()
+    code = 12345
+    UserResetPasswordCode.objects.create_code(user=test_user, code=12345)
+    res = client.post(
+        f"/api/password/reset/code/confirm/{test_user.email}/", {"code": code, "password": "testing12345"}, **JSON_RQST_HEADERS
+    )
+    assert res.status_code == 200
+    assert res.json()
+    res = client.post("/api/login/", {"email": test_user.email, "password": "testing12345"}, **JSON_RQST_HEADERS)
+    assert res.status_code == 200
+    assert res.json()
```

Comment:
Random aside. We need to get away from 200 checks.
Because of reasons, all responses will be a 200, so if you mistype the URL and are in a 404 condition, the status_code will still be 200.

So it's really the response payload that matters. I'm not sure but simply asserting that there is a json payload might be enough to avoid most issues we've seen with this.

Reply:
This is true and we may want to change that.
However, the default 404 page returned is not JSON and the `res.json()` would fail.
So I guess the test would work either way?

Reply:
Yea @oudeismetis is right I think we moved away from doing that! I might have copied this from another project. 
@NC-Fran  The response would not be 404 if the route does not exist is what Ed is saying.

Reply:
I think the response does not have a 404 status code, but still is a 404 page which when you try to convert into json, fails

Reply:
yea exactly, I assumed checking the json object would work.
I believe for SPA apps Google's crawler now inspects the page contents to see if it's displaying a 404 message.
Kinda feels like breaking the internet but 😐 

Either way...we can set a good example here in the bootstrapper with how we test and then collectively try and educate other devs on avoiding "200 only" assert statements on other projects.